plugin/adsk/scripts/USDMenuProc.mel

```diff
@@ -13,21 +13,43 @@
 // limitations under the License.
 //
 
-proc int canEditAsMaya(string $obj)
+proc int canEditAsMaya()
 {
     if (!hasPrimUpdater())
         return 0;
 
-    if (size($obj) == 0) {
-        string $nonMayaObjs[] = `python("import maya.internal.ufeSupport.utils as ufeUtils; ufeUtils.getNonMayaSelectedItems()")`;
-        $obj = $nonMayaObjs[0];
-    }
+    string $nonMayaObjs[] = `python("import maya.internal.ufeSupport.utils as ufeUtils; ufeUtils.getNonMayaSelectedItems()")`;
+    string $obj = $nonMayaObjs[0];
     if (size($obj) != 0) {
         return `python("from mayaUsd.lib import PrimUpdaterManager; PrimUpdaterManager.canEditAsMaya('" + $obj + "')")`;
     }
     return 0;
 }
 
+global proc mayaUsdMenu_editAsMaya()
+{
+    if (!hasPrimUpdater())
+        return;
+
+    string $nonMayaObjs[] = `python("import maya.internal.ufeSupport.utils as ufeUtils; ufeUtils.getNonMayaSelectedItems()")`;
+    string $obj = $nonMayaObjs[0];
+    if (size($obj) != 0) {
+        mayaUsdEditAsMaya $obj;
+    }
```

Comment:
The title of this PR says "edit-as-maya **entire** maya selection". The code you have written does not do that, it only edits the **first** non-Maya selected item? How is that different than using the input $obj to `USDMenuProc`?

Reply:
Well, I'm describing what the observable results of the use-case are. Before my changes, if a multi-part object was selected by selecting its root node, the entire object geometry would be highlighted in the viewport but right-clicking on it would only edit the sub-object under the mouse.

With my changes, Edit-as-Maya edits starting from the root node that was selected.

I'm not claiming anything about the multi-objects case, as we don't support invoking edit-as-Maya on multiple objects at once. Note that the old code was also only editing the first object.

A case could be made to issue multiple Edit-as-Maya commands if multiple objects are selected, but that would be a separate story.

Reply:
Note: this is all based on the pre-requisite that USD selection is in sub-component mode. When in component mode, the old code would edit the whole component. In sub-component mode, it would edit only a ingle sub-component, even though the selection was on a parent component.

Reply:
Note #2: if you got multiple objects selected, right-clicking does not show any menu, but with the old and new code.

Reply:
I changed the PR title to better reflect the intent. The JIRA issue was that when an object with sub-component was selected, right-clicking on it would only edit a sub-part, the not the whole selected object.

Reply:
Let's have a discussion offline. I looked at the original bug logged and following the steps I don't need any sub-component mode to see the problem. I can also select an item and then RMB over a completely different non-selected item and that one will be the input $obj (even though its not selected).

Reply:
I modified the code to use the object if it is not under the selection.

If it is under the selection, then I use the selection. (Well the first object in the selection as Edit-as-Maya only takes one object.)

Reply:
I retested with selection, without, and object outside the selection:

- RMB on object outside the selection: edits the object, ignores selection.
- RMB on sub-object of the selection: edits the first object of the selection instead of the sub-object.
- RMB without a selection: edits the object.

For the last one, the old code would not even show the menu...

Reply:
Not sure if this is useful or relevant, but Ufe::Selection has a containsAncestor() member.

Reply:
In my case, I would need containsDescendant()

Reply:
Which is there as well, also O(m) for a path of m components.

Reply:
Ah unfortunately, given the discussion, I actually need to find *which* item in the selection is the ancestor. So while containsAccestorInclusive) did the right thing during my testing, there is no function to retrieve that ancestor, so I will have to loop over the items in the selection anyway.

Reply:
I would do it differently: I would loop on the path components, calling Ufe::Selection::contains() for each path.  This will be better performing, because paths are typically short, and for large selections this will be a win.

Reply:
What I did is first check if the selection is an ancestor (quick check). Then for each element of the selection, I check if this is the ancestor.

Reply:
I mostly implemented what you suggested, thanks for all the input, using the right function really simplified the code. (And is very probably faster with large selections.)